tools/ashlar/ashlar.xml

```diff
@@ -0,0 +1,184 @@
+<tool id="ashlar" name="ASHLAR" version="@VERSION@.1" profile="17.09">
+    <description>Alignment by Simultaneous Harmonization of Layer/Adjacency Registration</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+ 
+    <expand macro="requirements"/>
+    @VERSION_CMD@
+
+    <command detect_errors="exit_code"><![CDATA[
+
+        #def clean(file,type)
+          #set name_clean = str($file.element_identifier).replace('.ome.tiff','').replace('.tiff','').replace('.tiff.','')
+
+          #if $type == "raw"
+            #set file_clean = $name_clean + ".ome.tiff"
+          #elif $type == "ffp"
+            #set file_clean = $name_clean + "_ffp.ome.tiff"
+          #elif $type == "dfp"
+            #set file_clean = $name_clean + "_dfp.ome.tiff"
+          #end if
+
+          #return $file_clean
+        #end def
+
+        ## Link the illumination files to appropriate file extension
+        #for $dfp in $ldfp:
+            ln -s "$dfp" "$clean($dfp,"dfp")" &&
+        #end for
+        #for $ffp in $lffp:
+            ln -s "$ffp" "$clean($ffp,"ffp")" &&
+        #end for
+
+        @CMD_BEGIN@
+
+        ## Supply the raw images
+        #for $raw in $lraw:
+            "$raw"
+        #end for
+
+        ## Additional arguments
+        -m $max_shift
+
+        #if $flip_x
+        --flip-x
+        #end if
+
+        #if $flip_y
+        --flip-y
+        #end if
+
+        -c $adv.align_channel
+
+        #if $adv.filter_sigma
+        --filter-sigma $adv.filter_sigma
+        #end if
+
+        #if $adv.tile_size
+        --tile-size $adv.tile_size
+        #end if
+
+        --ffp
+        #for $ffp in $lffp:
+            "$clean($ffp,"ffp")"
+        #end for
+
+        --dfp
+        #for $dfp in $ldfp:
+            "$clean($dfp,"dfp")"
+        #end for
+
+        #if $adv.pyramid
+        --pyramid
+        #end if
+
+        #if $adv.flip_mosaic_x
+        --flip-mosaic-x
+        #end if
+
+        #if $adv.flip_mosaic_y
```

Comment:
You can simplify booleans a lot. If you add the the param description below something like `truevalue="--flip-mosaic-y" falsevalue=""` you can write here just `$adv.flip_mosaic_y` and remove the entire "if"